### PR TITLE
Attack Proc fixes

### PR DESCRIPTION
**Pull request by NanakoAC**
Baymerge changed the attack proc, i think by removing a default value, and a lot of things broke because they weren't using the function correctly.

This proc does three main things.

1. Adds a default to /obj/item/proc/attack to target the chest when no target zone is supplied.

2. Fixes the lack of a target zone in many attack procs, properly defining it so it'll get passed to the parent, and ensure that bodypart-specific targeting works.

3. Removes most instances of code to retrieve the user's selected zone, and instead to use the zone passed into the attack function. saves duplication of effort

This fixes toolboxes not doing damage (they weren't passing a target zone)

Also vaguely related, also fixes nymphs being immortal, they were lacking a call to parent in their attackby proc

There's nothing controversial here, but it should be reviewed carefully for typos or mistakes due to the large volume of things changed

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 1]
```diff
@@ -152,9 +152,11 @@ emp_act
 
 /mob/living/carbon/human/resolve_item_attack(obj/item/I, mob/living/user, var/target_zone)
 	if(check_attack_throat(I, user))
+		world << "Attack throat is true?"
```
Comment (Arrow768) [inline code/modules/mob/living/carbon/human/human_defense.dm]: Forgotten Debug Messages ?

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 2]
```diff
@@ -152,9 +152,11 @@ emp_act
 
 /mob/living/carbon/human/resolve_item_attack(obj/item/I, mob/living/user, var/target_zone)
 	if(check_attack_throat(I, user))
+		world << "Attack throat is true?"
 		return null
 
 	if(user == src) // Attacking yourself can't miss
+		world << "Returning target zone"
```
Comment (Arrow768) [inline code/modules/mob/living/carbon/human/human_defense.dm]: Forgotten Debug Messages ?

**Comment by NanakoAC** [comment]
fixed

**Comment by NanakoAC** [comment]
the runtime error on this is caused by the cargo stock thing, and is unrelated to anything in this PR

File: code/game/objects/items/crayons.dm [hunk 3]
```diff
@@ -102,4 +102,4 @@
 				user << "<span class='warning'>You ate your crayon!</span>"
 				qdel(src)
 	else
-		..()
+		..(M, user, target_zone)
```
Comment (skull132) [inline code/game/objects/items/crayons.dm]: Unnecessary, should be reverted to `..()`

File: code/game/objects/items/weapons/storage/toolbox.dm [hunk 4]
```diff
@@ -96,12 +96,12 @@
...
-/obj/item/weapon/storage/toolbox/attack(mob/living/M as mob, mob/user as mob)
+/obj/item/weapon/storage/toolbox/attack(mob/living/M as mob, mob/user as mob, var/target_zone)
 	update_force()
-	..(M, user)
-	if (contents.len)
-		spill(3, get_turf(M))
-		playsound(M, 'sound/items/trayhit2.ogg', 100, 1)  //sound playin' again
-		update_force()
-		user.visible_message(span("danger", "[user] smashes the [src] into [M], causing it to break open and strew its contents across the area"))
+	if (..())
```
Comment (skull132) [inline code/game/objects/items/weapons/storage/toolbox.dm]: This is fine. Though you can, again, remove `var/target_zone` from the proc prototype.

**Pull request merged**